src/overlays/actors/ovl_En_Ishi/z_en_ishi.c

```diff
@@ -521,82 +546,87 @@ void func_8095E660(EnIshi* this, PlayState* play) {
+        EnIshi_SetupThrown(this);
+        EnIshi_ApplyGravity(this);
+        EnIshi_SetVelocity(&this->actor.velocity, sIshiVelocities[ENISHI_GET_BIG_FLAG(&this->actor)]);
         Actor_UpdatePos(&this->actor);
         Actor_UpdateBgCheckInfo(play, &this->actor, 7.5f, 35.0f, 0.0f,
                                 UPDBGCHECKINFO_FLAG_1 | UPDBGCHECKINFO_FLAG_4 | UPDBGCHECKINFO_FLAG_40 |
                                     UPDBGCHECKINFO_FLAG_80);
-    } else {
-        sp30.x = this->actor.world.pos.x;
-        sp30.y = this->actor.world.pos.y + 20.0f;
-        sp30.z = this->actor.world.pos.z;
+    } else { // still being held, follow player
+        pos.x = this->actor.world.pos.x;
+        pos.y = this->actor.world.pos.y + 20.0f;
+        pos.z = this->actor.world.pos.z;
+        // why does it track floor height? it uses bgcollision to detect floor hit in Thrown
         this->actor.floorHeight =
-            BgCheck_EntityRaycastFloor5(&play->colCtx, &this->actor.floorPoly, &bgId, &this->actor, &sp30);
+            BgCheck_EntityRaycastFloor5(&play->colCtx, &this->actor.floorPoly, &bgId, &this->actor, &pos);
     }
 }
 
-void func_8095EA70(EnIshi* this) {
-    f32 sp24;
+void EnIshi_SetupThrown(EnIshi* this) {
+    f32 randomStart;
 
     this->actor.velocity.x = Math_SinS(this->actor.world.rot.y) * this->actor.speed;
     this->actor.velocity.z = Math_CosS(this->actor.world.rot.y) * this->actor.speed;
-    if (!ENISHI_GET_1(&this->actor)) {
-        sp24 = Rand_ZeroOne() - 0.9f;
-        D_8095F690 = sp24 * 11000.0f;
-        D_8095F694 = ((Rand_ZeroOne() - 0.5f) * 3000.0f) * (fabsf(sp24) + 0.1f);
+    if (!ENISHI_GET_BIG_FLAG(&this->actor)) {
+        randomStart = Rand_ZeroOne() - 0.9f;
+        sIshiThrownXRotationVel = randomStart * 11000.0f;
+        sIshiThrownYRotationVel = ((Rand_ZeroOne() - 0.5f) * 3000.0f) * (fabsf(randomStart) + 0.1f);
     } else {
-        sp24 = Rand_ZeroOne() - 0.5f;
-        D_8095F690 = sp24 * 6000.0f;
-        D_8095F694 = ((Rand_ZeroOne() - 0.5f) * 1200.0f) * (fabsf(sp24) + 0.5f);
+        randomStart = Rand_ZeroOne() - 0.5f;
+        sIshiThrownXRotationVel = randomStart * 6000.0f;
+        sIshiThrownYRotationVel = ((Rand_ZeroOne() - 0.5f) * 1200.0f) * (fabsf(randomStart) + 0.5f);
     }
+
     this->actor.colChkInfo.mass = 200;
-    this->unk_194 = 100;
-    this->actionFunc = func_8095EBDC;
+    this->thrownTimer = 100;
+    this->actionFunc = EnIshi_Thrown;
 }
 
-void func_8095EBDC(EnIshi* this, PlayState* play) {
+void EnIshi_Thrown(EnIshi* this, PlayState* play) {
     s32 pad;
-    s32 sp70 = ENISHI_GET_1(&this->actor);
-    s16 temp_s0;
+    s32 isBig = ENISHI_GET_BIG_FLAG(&this->actor);
+    s16 pad2;
     s32 i;
-    s16 phi_s0;
-    Vec3f sp58;
-    s32 temp_v0 = (this->collider.base.atFlags & AT_HIT) != 0;
+    s16 spashAngle;
+    Vec3f pos;
+    s32 ATFlagSet = (this->collider.base.atFlags & AT_HIT) != 0;
```

Comment:
```suggestion
    s32 colliderATHit = (this->collider.base.atFlags & AT_HIT) != 0;
```
To mirror what you did above with `colliderACHit`